components/collector/src/base_collectors/source_collector.py

```diff
@@ -219,31 +219,72 @@ async def _landing_url(self, responses: SourceResponses) -> URL:
         return URL(url.removesuffix("xml") + "html" if url.endswith(".xml") else url)
 
 
-class UnmergedBranchesSourceCollector(SourceCollector, ABC):
-    """Base class for unmerged branches source collectors."""
+class BranchType(TypedDict):
```

Comment:
The information in this class and its children seems to resemble something more like `BranchInfo` or `BranchMeta`

Reply:
It's called `BranchType` because it's the type of the `Branch` typevar:

```python
class InactiveBranchesSourceCollector[Branch: BranchType](SourceCollector, ABC):
    """Base class for inactive branches source collectors."""
```

This syntax means that the `Branch` typevar is restricted to be a subtype of `BranchType`. See https://mypy.readthedocs.io/en/stable/generics.html#type-variables-with-upper-bounds.

I guess I could rename `BranchType` to `BranchTypeVarType` but not sure that's more clear, lol.